src/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        public unsafe struct LinkLayerAddressInfo
+        {
+            public int InterfaceIndex;
+            public fixed byte AddressBytes[8];
+            public byte NumAddressBytes;
+            private byte __pading;
```

Comment:
I would expect the compiler to properly determine padding and struct alignment for structs. From our experience, manual padding or struct packing is a source of AVs. The padding will have different behaviors on 32 vs 64 bit architectures.

Did you mean a different type of LayoutKind or is there a bug in the cross-compiler for the intended platform?